### DISCUSSION TITLE
How to have a layout with multiple chrome windows, each loading a given URL?

### DISCUSSION THREAD

**Discussion by haschdl** [rxhanson/RectanglePro-Community#347] [Q&A]
Is this possible? I tried multiple combinations of search words (page, webpage, URL, chrome...) and couldn't find an answer.
Is there a way (UI or scripted) to have a layout to be a combination of window positions + a given web page URL?
My use-case is studying, in particular studying a new language. I have 4 pages that I need to open *every single time* I start studying (at lest 6 times a week): the course material page (half left, display 2), google translate A=>B (top-right, display 2), google translate B=>A (bottom-right, display 2), Notion app (Full screen, display 1) - you get the gist.
 Rectangle Pro user, MacOS

**Comment by rxhanson** [comment]
Thanks for searching prior to asking. There isn't really a way to do this at this time. You can setup an app layout to load all of the chrome windows (although the fullscreen one would be maximized size in the layout, not "full screened"). Being able to specify URLs to load would require some scripting behavior to be built into the app to allow something like executing cmd+L and entering a specified URL. This starts to reach outside the scope of what I would like to add to to Rectangle Pro, since it's more about window management and not as much about scripting, but I'm always open to ideas, particularly if there is strong demand.

**Reply by michaelherger** [reply]
Is this the same issue I'm trying to resolve? I have two Chrome windows. And I want one to be on the center screen, full size, the other one third left on a secondary screen. I was able to save an app layout with all my apps. But the two Chrome windows are arranged "randomly". Sometimes it's ok, sometimes the main window would end up in the left third and vice versa.

**Reply by rxhanson** [reply]
In macOS, there aren't distinguishing characteristics for different windows of the same app that Rectangle Pro can properly leverage. It works by just applying each sequential layout to each window in the z-indexed order provided by macOS. So, the frontmost window will get the first layout, and so on. I've noticed that sometimes the z-indexed order provided by macOS isn't necessarily consistent, so there might be some inherent randomness here. It's on my todo list to revisit this and see if there are any other characteristics I can pull out that I had overlooked before.

**Reply by michaelherger** [reply]
Got it, thanks! I think the Z indexed order might be helpful, as this means it isn't really random. That order usually stays the same most of the time.

**Comment by haschdl** [comment]
thanks @rxhanson!  I realized my request might be not in the core feature of Rectangle Pro right now, but I think it makes more sense for certain workflows that essentially around web apps - for example if you're office worker using Google Suite, you might spend most of your time using various "apps" in the browser, switching tabs between email, slides, docs etc. So something to consider!
Meanwhile, I was able to prototype a AppleScript that is closer to what I need - adding link here, maybe others can find useful.

```applescript
-- List of URLs and layout keys

set urlLayoutList to {¬
	{URL:"https://nok.se", layout:"left-half"}, ¬
	{URL:"https://translate.google.com/?sl=en&tl=sv", layout:"top-right"}, ¬
	{URL:"https://translate.google.com/?sl=sv&tl=en", layout:"bottom-right"} ¬
		}

repeat with i from 1 to length of urlLayoutList
	set urlLayout to item i of urlLayoutList
	set urlItem to URL of urlLayout
	set layoutItem to layout of urlLayout
	-- display dialog "Page: " & urlItem & " Layout: " & layoutItem
	
	tell application "Safari"
		make new document with properties {URL:urlItem}
		activate
		delay 1
	end tell
	set layoutStr to "rectangle-pro://execute-action?name=" & layoutItem
	do shell script "open -g \"" & layoutStr & "\""
end repeat
```
https://gist.github.com/haschdl/0187b9068c27f16a7b731cc67b0ff827
